File: Data Structures & Algorithms/binary-tree-inorder-traversal/submission-1.cs
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     public int val;
 *     public TreeNode left;
 *     public TreeNode right;
 *     public TreeNode(int val=0, TreeNode left=null, TreeNode right=null) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
public class Solution {

    public List<int> InorderTraversal(TreeNode root) 
    {
        List<int> result = new List<int>();

        void Dfs(TreeNode root) 
        {
            if (root == null)
                return;
            Dfs(root.left);
            result.Add(root.val);
            Dfs(root.right);
        }

        Dfs(root);

        return result;
    }
}

File: Data Structures & Algorithms/path-sum/submission-1.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def hasPathSum(self, root: Optional[TreeNode], targetSum: int) -> bool:

        curr_sum = 0

        def backtrack(root: Optional[TreeNode], curr_sum: int, targetSum: int):

            if not root:
                return False            
            curr_sum += root.val
            
            if not root.left and not root.right and curr_sum == targetSum:
                return True
            if backtrack(root.left, curr_sum, targetSum):
                return True
            if backtrack(root.right, curr_sum, targetSum):
                return True

            curr_sum -= root.val
            return False
        
        result = backtrack(root, curr_sum, targetSum)
        return result
